File: backend/core/src/modules/characters/characters.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { ProjectScopedCrudController } from '../../common/access/project-scoped-crud.controller';
import { CharactersService } from './characters.service';
import { Character } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('characters')
export class CharactersController extends ProjectScopedCrudController<Character> {
  constructor(service: CharactersService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/CharactersController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudController;
import store.pjcloud.cowrite.core.entity.Character;
import store.pjcloud.cowrite.core.service.CharactersService;

@RestController
@RequestMapping("/characters")
public class CharactersController extends ProjectScopedCrudController<Character> {
    public CharactersController(CharactersService service) {
        super(service);
    }
}
